src/NServiceBus.FileBasedRouting/XmlRoutingFileParser.cs

```diff
@@ -42,23 +45,32 @@ public IEnumerable<EndpointRoutingConfiguration> Parse(XDocument document)
             return configs;
         }
 
-        static IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
+        IEnumerable<Type> GetAllMessageTypes(XElement endpointElement, string singular, string plural)
         {
             var handles = endpointElement.Element("handles");
 
             var separatelyConfiguredCommands = handles
                 ?.Elements(singular)
                 .Select(e => SelectCommand(e.Attribute("type").Value))
+                .Where(e => e != null)
```

Comment:
I think ideally (not sure if this is possible in a reliable way) for a given type `t` we should check if we know it's assembly. If not, we ignore such a type without logging warning. We would only warn if we know the assembly name is fine and we have it loaded but the type name is not recognized. 

We risk that a route with an invalid assembly name will be silently ignored but I think it might be better than logging `warn` for each unknown type if the message contracts are spread across many assemblies.